watertap/unit_models/_reverse_osmosis_base.py

```diff
@@ -749,10 +752,13 @@ def eq_N_Sh(b, t, x):
             )
             def eq_N_Sc(b, t, x):
                 bulk = b.feed_side.properties[t, x]
+                # # TODO: This needs to be revisted. Diffusion is now by component, but
+                #   not H2O and this var should also be by component, but the implementation
+                #   is not immediately clear.
                 return (
                     b.N_Sc[t, x]
                     * bulk.dens_mass_phase["Liq"]
-                    * bulk.diffus_phase["Liq"]
+                    * bulk.diffus_phase_comp["Liq", bulk.params.component_list.last()]
```

Comment:
Yes, this should be fixed so that the constraint `eq_N_Sc` is also indexed by solute_set or ion_set as component `j`. I _think_ the NF DSPM-DE model should have this implemented correctly.

Reply:
However, since we've only ever used the RO model for either the NaCl or seawater prop packs, we never had to worry about more than one solute, and I think we even have a warning or possibly an exception indicating the model is only compatible with one solute. Thus, this change will suffice for now.